### PR TITLE
Add/Update AWS Audit Manager > GxP 21 CFR Part 11. Closes #598 

### LINKED ISSUE #598: Add/Update AWS Audit Manager > GxP 21 CFR Part 11
**Is your feature request related to a problem? Please describe.**

**New Controls**

Benchmark | Control(s)
-- | --
11_10 Controls for closed systems | db_instance_backup_enabled
  | dynamodb_in_backup_plan
  | dynamodb_pitr_enabled
  | ebs_in_backup_plan
  | ebs_optimized_instance
  | efs_in_backup_plan
  | elasticache_redis_cluster_automatic_backup_check
  | rds_in_backup_plan
  | redshift_backup_enabled
  | s3_bucket_replication_enabled
  | s3_bucket_versioning_enabled
11_10_a | aurora_resources_protected_by_backup_plan
  | backup_plan_min_frequency_and_min_retention_check
  | backup_recovery_point_encrypted
  | backup_recovery_point_manual_deletion_disabled
  | backup_recovery_point_minimum_retention_check
  | cloudtrail_security_trail_enabled
  | dynamodb_resources_protected_by_backup_plan
  | ebs_resources_protected_by_backup_plan
  | ec2_resources_protected_by_backup_plan
  | ec2_volume_inuse_check
  | efs_resources_protected_by_backup_plan
  | elb_deletion_protection_enabled
  | fsx_resources_protected_by_backup_plan
  | rds_resources_protected_by_backup_plan
11_10_c | cloud_trail_enabled
  | cloud_trail_log_file_validation_enabled
  | db_instance_backup_enabled
  | dynamodb_in_backup_plan
  | dynamodb_pitr_enabled
  | ebs_in_backup_plan
  | ebs_optimized_instance
  | ecr_private_lifecycle_policy_configured
  | efs_in_backup_plan
  | elasticache_redis_cluster_automatic_backup_check
  | rds_in_backup_plan
  | redshift_backup_enabled
  | s3_bucket_replication_enabled
  | s3_lifecycle_policy_check
  | s3_version_lifecycle_policy_check
11_10_d | dynamodb_table_encrypted_kms
  | ec2_ebs_encryption_by_default
  | efs_encrypted_check
  | elasticsearch_encrypted_at_rest
  | elasticsearch_node_to_node_encryption_check
  | encrypted_volumes
  | iam_role_managed_policy_check
  | restricted_incoming_traffic
  | s3_account_level_public_access_blocks
  | s3_bucket_level_public_access_prohibited
  | s3_bucket_policy_grantee_check
  | ssm_document_not_public
  | subnet_auto_assign_public_ip_disableds3_account_level_public_access_blocks_periodic
11_10_e | db_instance_backup_enabled
  | dynamodb_in_backup_plan
  | dynamodb_pitr_enabled
  | ebs_in_backup_plan
  | ebs_optimized_instance
  | efs_in_backup_plan
  | elasticache_redis_cluster_automatic_backup_check
  | opensearch_audit_logging_enabled
  | opensearch_logs_to_cloudwatch
  | rds_in_backup_plan
  | redshift_audit_logging_enabled
  | redshift_backup_enabled
  | s3_bucket_replication_enabled
  | s3_bucket_versioning_enabled
11_10_g | efs_encrypted_check
  | restricted_incoming_traffic
  | s3_account_level_public_access_blocks
  | s3_account_level_public_access_blocks_periodic
  | s3_bucket_level_public_access_prohibited
  | s3_bucket_policy_grantee_check
  | ssm_document_not_public
11_10_k | autoscaling_launch_config_public_ip_disabled
  | restricted_incoming_traffic
  | s3_account_level_public_access_blocks
  | ssm_document_not_public
11_200_a | access_keys_rotated
  | iam_password_policy
  | iam_root_access_key_check
  | iam_user_mfa_enabled
  | mfa_enabled_for_iam_console_access
  | root_account_hardware_mfa_enabled
  | root_account_mfa_enabled
11_30 Controls for open systems | api_gw_ssl_enabledelbv2_acm_certificate_required
  | backup_recovery_point_encrypted
  | cloudfront_no_deprecated_ssl_protocols
  | cloudfront_traffic_to_origin_encrypted
  | codebuild_project_artifact_encryption
  | codebuild_project_s3_logs_encrypted
  | efs_encrypted_check
  | kinesis_stream_encrypted
  | opensearch_encrypted_at_rest
  | opensearch_https_required
  | opensearch_node_to_node_encryption_check

**Removed Controls**

Benchmark | Control(s)
-- | --
11_10_a | ebs_volume_unused
11_10_a | elb_application_lb_deletion_protection_enabled
11_10_c | redshift_cluster_kms_enabled
11_10_d | s3_public_access_block_account
11_10_d | s3_public_access_block_bucket
11_10_d | vpc_security_group_restrict_ingress_common_ports_all
11_10_d | vpc_subnet_auto_assign_public_ip_disabled
11_10_e | redshift_cluster_kms_enabled
11_10_g | efs_file_system_encrypt_data_at_rest
11_10_g | s3_public_access_block_account
11_10_g | s3_public_access_block_bucket
11_10_g | vpc_security_group_restrict_ingress_common_ports_all
11_10_k | vpc_security_group_restrict_ingress_common_ports_all
11_200 Electronic signature components and controls | iam_account_password_policy_strong
11_200 | iam_root_user_hardware_mfa_enabled
11_200 | iam_root_user_mfa_enabled
11_200 | iam_root_user_no_access_keys
11_200 | iam_user_access_key_age_90
11_200 | iam_user_console_access_mfa_enabled
11_200 | iam_user_mfa_enabled
11_30 Controls for open systems | apigateway_rest_api_stage_use_ssl_certificate
11_30 | efs_file_system_encrypt_data_at_rest
11_30 | elb_application_network_lb_use_ssl_certificate
11_30 | rds_db_instance_encryption_at_rest_enabled

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by madhushreeray30**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: gxp_21_cfr_part_11/gxp_21_cfr_part_11.sp [hunk 1]
```diff
@@ -11,8 +11,8 @@ benchmark "gxp_21_cfr_part_11" {
   documentation = file("./gxp_21_cfr_part_11/docs/gxp_21_cfr_part_11_overview.md")
   children = [
     benchmark.gxp_21_cfr_part_11_11_10,
-    benchmark.gxp_21_cfr_part_11_11_30,
     benchmark.gxp_21_cfr_part_11_11_200,
+    benchmark.gxp_21_cfr_part_11_11_30,
```
Comment (cbruno10) [inline gxp_21_cfr_part_11/gxp_21_cfr_part_11.sp:15]: @madhushreeray30 Why was this moved? In GxP's organization, does 11_11_30 come after 11_11_200?

File: gxp_21_cfr_part_11/11_200.sp [hunk 2]
```diff
@@ -1,8 +1,18 @@
 benchmark "gxp_21_cfr_part_11_11_200" {
   title       = "11.200 Electronic signature components and controls"
+  description = "TO DO"
```
Comment (cbruno10) [inline gxp_21_cfr_part_11/11_200.sp:3]: What is this supposed to be?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@madhushreeray30 I left some comments, can you please have a look? Thanks!

File: gxp_21_cfr_part_11/11_200.sp (hunk 2 above)
Comment (cbruno10) [reply gxp_21_cfr_part_11/11_200.sp:3]: Do we have any description we can use? Will the Hub page look blank/empty (assming we don't have a separate doc for this one)?

File: conformance_pack/codebuild.sp [hunk 3]
```diff
@@ -93,11 +93,24 @@ control "codebuild_project_environment_privileged_mode_disabled" {
 
 control "codebuild_project_artifact_encryption_enabled" {
   title       = "CodeBuild project artifact encryption should be enabled"
-  description = "This control checks if an AWS CodeBuild project has encryption enabled for all of its artifacts. The rule is non compliant if 'encryptionDisabled' is set to 'true' for any primary or secondary (if present) artifact configurations."
+  description = "This control checks if an CodeBuild project has encryption enabled for all of its artifacts. The rule is non compliant if 'encryptionDisabled' is set to 'true' for any primary or secondary (if present) artifact configurations."
```
Comment (misraved) [inline conformance_pack/codebuild.sp:96]:
```suggestion
  description = "This control checks if a CodeBuild project has encryption enabled for all of its artifacts. The rule is non compliant if 'encryptionDisabled' is set to 'true' for any primary or secondary (if present) artifact configurations."
```

File: conformance_pack/opensearch.sp [hunk 4]
```diff
@@ -52,8 +58,21 @@ control "opensearch_domain_logs_to_cloudwatch" {
...
+    gxp_21_cfr_part_11 = "true"
+    gxp_eu_annex_11    = "true"
+    pci_dss_v321       = "true"
+    soc_2              = "true"
+  })
+}
+
+control "opensearch_domain_node_to_node_encryption_enabled" {
+  title       = "OpenSearch domains node-to-node encryption should be enabled"
+  description = "This control check if Amazon OpenSearch Service nodes are encrypted end to end. The rule is non compliant if the node-to-node encryption is not enabled on the domain."
```
Comment (misraved) [inline conformance_pack/opensearch.sp:70]:
```suggestion
  description = "This control checks if Amazon OpenSearch Service nodes are encrypted end to end. The rule is non-compliant if the node-to-node encryption is not enabled on the domain."
```

File: conformance_pack/s3.sp [hunk 5]
```diff
@@ -311,8 +312,9 @@ control "s3_bucket_versioning_and_lifecycle_policy_enabled" {
   query       = query.s3_bucket_versioning_and_lifecycle_policy_enabled
 
   tags = merge(local.conformance_pack_s3_common_tags, {
-    pci_dss_v321 = "true"
-    soc_2        = "true"
+    pci_dss_v321       = "true"
+    gxp_21_cfr_part_11 = "true"
+    soc_2              = "true"
```
Comment (misraved) [inline conformance_pack/s3.sp:317]: Please sort the tags.

File: conformance_pack/s3.sp [hunk 6]
```diff
@@ -327,6 +329,16 @@ control "s3_bucket_event_notifications_enabled" {
   })
 }
 
+control "s3_bucket_policy_restrict_public_access" {
+  title       = "S3 bucket policy should prohibit public access"
+  description = "This control checks that the access granted by the S3 bucket is restricted by any of the principals, federated users, service principals, IP addresses, or VPCs that you provide. The rule is compliant if a bucket policy is not present."
```
Comment (misraved) [inline conformance_pack/s3.sp:334]:
```suggestion
  description = "This control checks that the access granted by the S3 bucket is restricted by any of the principals, federated users, service principals, IP addresses, or VPCs that you provide. The rule is non-compliant if a bucket policy is not present."
```
Should the description be updated to use non-compliant?

File: conformance_pack/s3.sp [hunk 7]
```diff
@@ -942,6 +954,28 @@ query "s3_bucket_event_notifications_enabled" {
...
+  sql = <<-EOQ
+    select
+      arn as resource,
+      case
+        when policy_std is null then 'ok'
+        when s ->> 'Effect' = 'Allow' and (s -> 'Principal' -> 'AWS' = '["*"]' or s ->> 'Principal' = '*') then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when policy_std is null then title || ' policy not publicly accessable.'
```
Comment (misraved) [inline conformance_pack/s3.sp:967]:
```suggestion
        when policy_std is null then title || ' policy not publicly accessible.'
```

File: conformance_pack/s3.sp [hunk 8]
```diff
@@ -942,6 +954,28 @@ query "s3_bucket_event_notifications_enabled" {
...
+        when s ->> 'Effect' = 'Allow' and (s -> 'Principal' -> 'AWS' = '["*"]' or s ->> 'Principal' = '*') then title || ' policy publicly accessable.'
```
Comment (misraved) [inline conformance_pack/s3.sp:968]:
```suggestion
        when s ->> 'Effect' = 'Allow' and (s -> 'Principal' -> 'AWS' = '["*"]' or s ->> 'Principal' = '*') then title || ' policy publicly accessible.'
```

File: conformance_pack/s3.sp [hunk 9]
```diff
@@ -942,6 +954,28 @@ query "s3_bucket_event_notifications_enabled" {
...
+        else title || ' policy not publicly accessable.'
```
Comment (misraved) [inline conformance_pack/s3.sp:969]:
```suggestion
        else title || ' policy not publicly accessible.'
```

File: conformance_pack/vpc.sp [hunk 10]
```diff
@@ -299,6 +298,16 @@ control "vpc_network_acl_unused" {
   })
 }
 
+control "vpc_security_group_allows_ingress_authorized_ports" {
+  title       = "VPC Security groups should only allow unrestricted incoming traffic for authorized ports"
+  description = "This control checks whether the vpc security groups that are in use allow unrestricted incoming traffic. Optionally the rule checks whether the port numbers are listed in the authorizedTcpPorts parameter. The default values for authorizedTcpPorts are 80 and 443."
```
Comment (misraved) [inline conformance_pack/vpc.sp:303]:
```suggestion
  description = "This control checks whether the VPC security groups that are in use allow unrestricted incoming traffic. Optionally the rule checks whether the port numbers are listed in the authorizedTcpPorts parameter. The default values for authorizedTcpPorts are 80 and 443."
```

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**